### PR TITLE
PP-640 Add events link in payment objects

### PR DESCRIPTION
**Pull request by imranbohoran**
**WHAT**
Introducing event links in payment objects. Every payment object will have a events link in the '_link' object. The only exception is the GET on payment events. As the commits explain there has been some restructuring and renaming of classes. Please also note that there's another part to the story where cancel links need to be added to payment objects.

**WHO**
any developer

### CODE REVIEW

**Comment by imranbohoran** [comment]
rebuild

File: swagger/swagger.json [hunk 1]
```diff
@@ -435,37 +488,6 @@
...
-        "self" : {
-          "description" : "self",
-          "readOnly" : true,
-          "$ref" : "#/definitions/paymentLink"
-        }
-      },
-      "description" : "Resource self link of a Payment"
     }
   }
 }
```
Comment (rauligs) [inline swagger/swagger.json]: Configure to add new lines automatically

**Comment by rauligs** [comment]
Get payment - check it does not contain events link

File: swagger/swagger.json (hunk 1 above)
Comment (imranbohoran) [inline swagger/swagger.json]: I checked and I have it setup in intelliJ. But given that this file is generated by the maven plugin and we don't hand-craft it this is what we get

**Comment by imranbohoran** [comment]
Re: test for GET payment. Thanks for that, it's added now.

**Pull request merged**